packages-ts/gauntlet-terra-contracts/src/lib/contracts.ts

```diff
@@ -28,109 +30,191 @@ export type TerraABI = {
-}
 
-export type Contracts = Record<CONTRACT_LIST, Contract>
+  constructor(id, dirName, defaultVersion) {
+    this.id = id
+    this.defaultVersion = defaultVersion
+    this.dirName = dirName
+  }
+
+  addInstance(envVar: string, name: string = this.id) {
+    const address = process.env[envVar]
+    if (!address) {
+      console.warn(`${envVar} not set in environment--ignoring`)
+      return this // it's okay if we don't have all the contract addresses, for now
+    }
+
+    if (!AccAddress.validate(address))
+      throw new Error(`Read invalid contract address ${address} for ${this.id} contract from env`)
 
-export const getContractCode = async (contractId: CONTRACT_LIST, version): Promise<string> => {
-  if (version === 'local') {
+    logger.debug(`Using deployed instance of ${this.id}: ${name}=${address}`)
+    return this
+  }
+
+  loadContractCode = async (version: string): Promise<void> => {
+    version = version ? version : this.defaultVersion
+    assertions.assert(
+      !this.version || version == this.version,
+      `Loading multiple versions (${this.version} and ${version}) of the same contract is unsupported.`,
+    )
+    this.version = version
+
+    if (version === 'local') {
+      // Possible paths depending on how/where gauntlet is being executed
+      const possibleContractPaths = [
+        path.join(__dirname, '../../artifacts/bin'),
+        path.join(process.cwd(), './artifacts/bin'),
+        path.join(process.cwd(), './tests/e2e/common_artifacts'),
+        path.join(process.cwd(), './packages-ts/gauntlet-terra-contracts/artifacts/bin'),
+      ]
+
+      const codes = possibleContractPaths
+        .filter((contractPath) => existsSync(`${contractPath}/${this.id}.wasm`))
+        .map((contractPath) => {
+          const wasm = readFileSync(`${contractPath}/${this.id}.wasm`)
+          return wasm.toString('base64')
+        })
+      this.bytecode = codes[0]
+    } else {
+      const url = `${this.downloadUrl}${version}/${this.id}.wasm`
+      logger.loading(`Fetching ${url}...`)
+      const response = await fetch(url)
+      const body = await response.arrayBuffer()
+      if (body.length == 0) {
+        throw new Error(`Download ${this.id}.wasm failed`)
+      }
+      this.bytecode = Buffer.from(body).toString('base64')
+    }
+  }
+
+  loadContractABI = async (): Promise<void> => {
     // Possible paths depending on how/where gauntlet is being executed
+    const cwd = process.cwd()
     const possibleContractPaths = [
-      path.join(__dirname, '../../artifacts/bin'),
-      path.join(process.cwd(), './artifacts/bin'),
-      path.join(process.cwd(), './tests/e2e/common_artifacts'),
-      path.join(process.cwd(), './packages-ts/gauntlet-terra-contracts/artifacts/bin'),
+      path.join(__dirname, './artifacts/contracts'),
+      path.join(cwd, './contracts'),
+      path.join(cwd, '../../contracts'),
+      path.join(cwd, './packages-ts/gauntlet-terra-contracts/artifacts/contracts'),
```

Comment:
The schemas for `cw-plus` contracts should be in `@chainlink/gauntlet-terra-cw-plus` package, not under `@chainlink/gauntlet-terra-contracts` as it is currently.